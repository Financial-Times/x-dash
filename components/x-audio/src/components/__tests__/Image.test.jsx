const { h } = require('@financial-times/x-engine');
const { shallow } = require('@financial-times/x-test-utils/enzyme');
const { Image } = require('../Image');

jest.mock('../classnames-helper', () => {
	const { classnamesHelperMock } = require('../../../__tests__/test-helper');
	return classnamesHelperMock;
});

describe('Image', () => {
	test('should have necessary elements', () => {
		const props = {
			imageDataSet: {
				url: 'image-url',
				resolutions: []
			}
		};
		const subject = shallow(<Image {...props}/>);

		expect(subject).toMatchSnapshot();
	});

	test('should set resolutions into srcSet correctly', () => {
		const props = {
			imageDataSet: {
				url: 'image-url',
				resolutions: [
					{ url: 'resolution-url-300px', resolution: '300px' },
					{ url: 'resolution-url-500px', resolution: '500px' }
				]
			}
		};
		const subject = shallow(<Image {...props}/>);

		expect(subject).toMatchSnapshot();
	});
});
