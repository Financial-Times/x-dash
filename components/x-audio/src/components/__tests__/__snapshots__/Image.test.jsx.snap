// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should have necessary elements 1`] = `
<div
  className="audio-player__info__image"
>
  <img
    alt=""
    src="image-url"
    srcSet=""
  />
</div>
`;

exports[`Image should set resolutions into srcSet correctly 1`] = `
<div
  className="audio-player__info__image"
>
  <img
    alt=""
    src="image-url"
    srcSet="resolution-url-300px 300px,resolution-url-500px 500px"
  />
</div>
`;
